src/arch/xtensa/hal/CMakeLists.txt

```diff
@@ -139,4 +139,5 @@ add_local_sources(hal
 	interrupts.c
 	memcopy.S
 	windowspill_asm.S
+	atomics.c
```

Comment:
I don't think this belongs to this place. As long as you are not using any xtensa cmpxchg type of instruction.

Maybe just add this inside src/lib/atomic.c ?